### PR TITLE
Compile and Run the Manifest on Windows

### PR DESCRIPTION
**Pull request by gmittert**
The swift interpreter doesn't work on Windows currently. Instead of
attempting to interpret the manifest, we compile it to a temporary file,
then execute it.

### CODE REVIEW

File: Sources/PackageLoading/ManifestLoader.swift [hunk 1]
```diff
@@ -419,7 +419,11 @@ public final class ManifestLoader: ManifestLoaderProtocol {
             }
           #endif
             cmd += [resources.swiftCompiler.pathString]
+            // The interpreter doesn't work on Windows yet, so instead
+            // we compile and run it on Windows
+#if !os(Windows)
```
Comment (aciidb0mb3r) [inline Sources/PackageLoading/ManifestLoader.swift]: Do you know if it'll be available soon? I think we should break this into cleaner functions for compiling vs interpreting and then always select the compiling one for windows.

Comment (gmittert) [reply Sources/PackageLoading/ManifestLoader.swift]: I don't think anyone is working on it at the moment. I can split them out though, that makes sense to me.

Comment (DougGregor) [reply Sources/PackageLoading/ManifestLoader.swift]: Using `#if os(Windows)` for this means the code is likely to break as others do refactoring an Darwin/Linux. Can it instead be some global `Bool` whose default value differs on Windows, but which will allow us to test the behavior on Darwin/Linux as well?

Comment (aciidb0mb3r) [reply Sources/PackageLoading/ManifestLoader.swift]: That's the plan (see my first comment).

**Comment by aciidb0mb3r** [comment]
This should be no longer needed with https://github.com/apple/swift-package-manager/pull/2518 Please open a new PR if some changes are still required for Windows support!

**Pull request closed without merging**